articles/200_migration_guide_from_ros1.md

```diff
@@ -162,3 +162,483 @@ NOTE: to be written
+
+~~~
+. /opt/ros/jade/setup.bash
+cd ~/ros1_talker
+catkin_make install
+~~~
+
+#### Running the ROS 1 node
+
+If there's not already one running, we start a `roscore`, first sourcing the
+setup file from our `catkin` install tree (the system setup file at
+`/opt/ros/jade/setup.bash` would also work here):
+
+~~~
+. ~/ros1_talker/install/setup.bash
+roscore
+~~~
+
+In another shell, we run the node from the `catkin` install space using
+`rosrun`, again sourcing the setup file first (in this case it must be the one
+from our workspace):
+ 
+~~~
+. ~/ros1_talker/install/setup.bash
+rosrun talker talker
+~~~
+
+### Migrating to ROS 2
+
+Let's start by creating a new workspace in which to work:
+
+~~~
+mkdir ~/ros2_talker
+cd ~/ros2_talker
+~~~
+
+We'll copy the source tree from our ROS 1 package into that workspace, where we can modify it:
+
+~~~
+mkdir src
+cp -a ~/ros1_talker/src/talker src
+~~~
+
+Now we'll modify the the C++ code in the node.  The ROS 2 C++ library, called
+`rclcpp`, provides a different API from that provided by `roscpp`.  The
+concepts are very similar between the two libraries, which makes the changes
+reasonably straightforward to make.
+
+#### Included headers
+
+In place of `ros/ros.h`, which gave us access to the `roscpp` library API, we
+need to include `rclcpp/rclcpp.hpp`, which gives us access to the `rclcpp`
+library API:
+
+~~~
+//#include "ros/ros.h"
+#include "rclcpp/rclcpp.hpp"
+~~~
+
+To get the `std_msgs/String` message definition, in place of
+`std_msgs/String.h`, we need to include `std_msgs/msg/string.hpp`:
+
+~~~
+//#include "std_msgs/String.h"
+#include "std_msgs/msg/string.hpp"
+~~~
+
+#### Changing C++ library calls
+
+Instead of passing the node's name to the library initialization call, we do
+the initialization, then pass the node name to the creation of the node object
+(we can use the `auto` keyword because now we're requiring a C++11 compiler):
+
+~~~
+//  ros::init(argc, argv, "talker");
+//  ros::NodeHandle n;
+    rclcpp::init(argc, argv);
+    auto node = rclcpp::node::Node::make_shared("talker");
+~~~
+
+The creation of the publisher and rate objects looks pretty similar, with some
+changes to the names of namespace and methods.  For the publisher, instead of an
+integer queue length argument, we pass a quality of service (qos) profile, which
+is a far more flexible way to controlling how message delivery is handled.  In
+this example, we just pass the default profile `rmw_qos_profile_default` (it's
+global because it's declared in `rmw`, which is written in C and so doesn't have
+namespaces).
+
+~~~
+//  ros::Publisher chatter_pub = n.advertise<std_msgs::String>("chatter", 1000);
+//  ros::Rate loop_rate(10);
+  auto chatter_pub = node->create_publisher<std_msgs::msg::String>("chatter",
+    rmw_qos_profile_default);
+  rclcpp::rate::Rate loop_rate(10);
+~~~
+
+The creation of the outgoing message is different in both the namespace and the
+fact that we go ahead and create a shared pointer:
```

Comment:
Should we mention that this is temporary too? Assuming that we will support publishing a const reference soon.